pre_commit/git.py

```diff
@@ -83,15 +84,16 @@ def get_files_matching(all_file_list_strategy):
     def wrapper(include_expr, exclude_expr):
         include_regex = re.compile(include_expr)
         exclude_regex = re.compile(exclude_expr)
-        return set(
-            filename
-            for filename in all_file_list_strategy()
-            if (
-                include_regex.search(filename) and
-                not exclude_regex.search(filename) and
-                os.path.lexists(filename)
+        with cwd(get_root()):  # because filenames can be relative
```

Comment:
It is.
I tested, and at this point in the code, when running the test I added, here is what I observe:
```
os.getcwd() : '/tmp/user/1000/pytest-2/test_invoke_precommit_in_subdi0/2/subdir'
all_file_list_strategy() : [u'subdir/foo.py']
```